akka-http-core/src/test/scala/akka/http/impl/util/AkkaSpecWithMaterializer.scala

```diff
@@ -41,13 +41,8 @@ abstract class AkkaSpecWithMaterializer(s: String)
 }
 object AkkaSpecWithMaterializer {
   // adapted version of AkkaSpec.getCallerName that also works for `AkkaSpecWithMaterializer`
-  def getCallerName(clazz: Class[_]): String = {
-    val s = (Thread.currentThread.getStackTrace map (_.getClassName) drop 1)
+  def callerName(): String =
+    (Thread.currentThread.getStackTrace map (_.getClassName) drop 1)
       .dropWhile(_ matches "(java.lang.Thread|.*AkkaSpecWithMaterializer.?$|.*StreamSpec.?$)")
-    val reduced = s.lastIndexWhere(_ == clazz.getName) match {
-      case -1 => s
-      case z  => s drop (z + 1)
-    }
-    reduced.head.replaceFirst(""".*\.""", "").replaceAll("[^a-zA-Z_0-9]", "_")
-  }
+      .head.replaceFirst(""".*\.""", "").replaceAll("[^a-zA-Z_0-9]", "_")
```

Comment:
Hard to see what we lose here (if anything)

Reply:
Maybe there were some setups where `getClass` worked differently but not here.